arrow/src/record_batch.rs

```diff
@@ -175,6 +175,12 @@ impl RecordBatch {
         self.schema.clone()
     }
 
+
+    /// Projects the schema onto the specified columns
+    pub fn project(&self, indices: impl IntoIterator<Item=usize>) -> Result<Schema> {
```

Comment:
The intent of this field was to project the `RecordBatch` rather than just the schema:

A signature like this:
```suggestion
    pub fn project(&self, indices: impl IntoIterator<Item=usize>) -> Result<RecordBatch> {
```

(so we would also have to project the columns as well as the schema)

Reply:
Ahh, I thought this part was a bit too easy, okay I'll update to reflect that.